src/createWebStoragePersistor-experimental/index.ts

```diff
@@ -16,10 +16,47 @@ export function createWebStoragePersistor({
   key = `REACT_QUERY_OFFLINE_CACHE`,
   throttleTime = 1000,
 }: CreateWebStoragePersistorOptions): Persistor {
+  //try to save data to storage
+  function trySave(persistedClient: PersistedClient) {
+    try {
+      storage.setItem(key, JSON.stringify(persistedClient))
+    } catch {
+      return false
+    }
+    return true
+  }
+
   if (typeof storage !== 'undefined') {
     return {
       persistClient: throttle(persistedClient => {
-        storage.setItem(key, JSON.stringify(persistedClient))
+        if (trySave(persistedClient) !== true) {
+          const mutations = [...persistedClient.clientState.mutations]
+          const queries = [...persistedClient.clientState.queries]
+          const client: PersistedClient = {
+            ...persistedClient,
+            clientState: { mutations, queries },
+          }
+
+          // clean mutations and try to save
+          while (mutations.shift()) {
+            if (trySave(client)) {
+              return // save success
+            }
+          }
```

Comment:
I think we shouldn't remove `paused` mutations - because those are the ones that are persisted so that they can be resumed later (e.g. when the user is online again).

see: https://github.com/tannerlinsley/react-query/blob/e6da990703482cff8ea99f5a254975cb57463ed5/src/core/mutationCache.ts#L115-L124

Reply:
All the mutations are `paused` as default.

In fact, I have no idea about this question: which is more important, the `queries` or the `mutations` 😄 .

Do you mean the persisted mutations are more important?
If we think so, we can try to clean the `queries ` at first.

Reply:
> All the mutations are paused as default.

not sure what you mean with that. A mutation can have one of the following states:

- `success`: it has been executed successfully
- `error`: it has been executed, but is erroneous
- `idle`: it has not been executed yet, or it has been reset
- `loading`: the mutation is currently running

Additionally, a mutation can be in `isPaused: true` "state" when it is running, but has been paused because the mutation cannot be retried because the network is offline.

I think it's very important to _not_ removed paused mutations, because they would likely be resumed when the network goes online again.

So, I think the order of removal should be:

- non-paused mutations
- queries
- paused mutations

Reply:
I found the `defaultShouldDehydrateMutation`
https://github.com/tannerlinsley/react-query/blob/69a5d798fed3beb69f015c131b05074050677223/src/core/hydration.ts#L68-L70

Reply:
aaah right, so per default, only paused mutations and successful queries are included in the hydration, but you can overwrite that. I still think the order above would be the most correct one, but it would mean that per default, there are no non-paused mutations, so we could maybe bypass that check.

However, it would still mean that queries should be removed before mutations I think, because mutations are likely _paused_, so removing them before successful queries seems wrong. What do you think?

Reply:
updated with tests in the 4th commit